### ISSUE TITLE
Problems with generalization performance on make3d datasets

### ISSUE THREAD

**Issue by GuanghuiWei** [noahzn/Lite-Mono#154]
Thank you very much for your contribution. I tried to generalize your weights directly to the make3d dataset, but I did not get the generalization effect in the paper. This is my generalization result, is there any problem?
![33333333](https://github.com/user-attachments/assets/de872a64-29f9-45ed-8c75-cadae28955e4)
![3333333333333333333](https://github.com/user-attachments/assets/5e977533-917b-42c7-b9b7-e54232f71e5e)

**Comment by noahzn** [comment]
Hi, I don't know what code you used for evaluation on make3d.

**Comment by GuanghuiWei** [comment]
![33322221111](https://github.com/user-attachments/assets/546f94e9-333a-4ce9-be6b-1aa9bac78e19)
This is the code I used to test make3d, testing 134 images. Please help me to see if there are any mistakes

**Comment by GuanghuiWei** [comment]
min_depth:0.1，max_depth:70

**Comment by noahzn** [comment]
Hi, I don't have the evaluation code with me now. Please take a look at the code [here](https://github.com/nianticlabs/monodepth2/issues/89)

**Comment by Wshenv** [comment]
`
from layers import disp_to_depth
import networks
import cv2
import os
import torch
from scipy import io
import numpy as np

load_weights_folder = '/path/to/weight/'
main_path = 'path/to/mask_3d/'
encoder_path = os.path.join(load_weights_folder, "encoder.pth")
decoder_path = os.path.join(load_weights_folder, "depth.pth")
encoder_dict = torch.load(encoder_path)
encoder = networks.LiteMono(model='lite-mono',
                            height=encoder_dict['height'],
                            width=encoder_dict['width'])
depth_decoder = networks.DepthDecoder(encoder.num_ch_enc, scales=range(3))

model_dict = encoder.state_dict()
encoder.load_state_dict({k: v for k, v in encoder_dict.items() if k in model_dict})
depth_decoder.load_state_dict(torch.load(decoder_path), strict=False)

encoder.cuda()
encoder.eval()
depth_decoder.cuda()
depth_decoder.eval()


def compute_errors(gt, pred):
    rmse = (gt - pred) ** 2
    rmse = np.sqrt(rmse.mean())

    rmse_log = (np.log10(gt) - np.log10(pred)) ** 2
    rmse_log = np.sqrt(rmse_log.mean())

    abs_rel = np.mean(np.abs(gt - pred) / gt)

    sq_rel = np.mean(((gt - pred) ** 2) / gt)

    return abs_rel, sq_rel, rmse, rmse_log


with open(os.path.join('/path/to/make3d_test/', "test_files.txt")) as f:
    test_filenames = f.read().splitlines()
    print(test_filenames)
# test_filenames = map(lambda x: x[4:], test_filenames)
# test_filenames = map(test_filenames)
# print(test_filenames)
depths_gt = []
images = []
ratio = 2
h_ratio = 1 / (1.33333 * ratio)
color_new_height = int(1704 / 2)
depth_new_height = 21
for filename in test_filenames:
    # print(filename)
    mat = io.loadmat(os.path.join(main_path, "Gridlaserdata", "depth_sph_corr-{}.mat".format(filename)))
    depths_gt.append(mat["Position3DGrid"][:, :, 3])

    image = cv2.imread(os.path.join(main_path, "Test134", "img-{}.jpg".format(filename)))
    image = image[int((2272 - color_new_height) / 2):int((2272 + color_new_height) / 2), :, :]
    images.append(image[:, :, ::-1])
depths_gt_resized = map(lambda x: cv2.resize(x, (305, 407), interpolation=cv2.INTER_NEAREST), depths_gt)
depths_gt_cropped = map(lambda x: x[int((55 - 21) / 2):int((55 + 21) / 2), :], depths_gt)

depths_gt_cropped = list(depths_gt_cropped)
errors = []
with torch.no_grad():
    for i in range(len(images)):
        input_color = images[i]
        input_color = cv2.resize(input_color / 255.0, (640, 192), interpolation=cv2.INTER_NEAREST)

        input_color = torch.tensor(input_color, dtype=torch.float).cuda().permute(2, 0, 1)[None, :, :, :]

        output = depth_decoder(encoder(input_color))
        pred_disp, _ = disp_to_depth(output[("disp", 0)], 1e-3, 70)
        pred_disp = pred_disp.squeeze().cpu().numpy()
        depth_gt = depths_gt_cropped[i]
        depth_pred = 1 / pred_disp
        depth_pred = cv2.resize(depth_pred, depth_gt.shape[::-1], interpolation=cv2.INTER_NEAREST)
        mask = np.logical_and(depth_gt > 0, depth_gt < 70)
        depth_gt = depth_gt[mask]
        depth_pred = depth_pred[mask]
        depth_pred *= np.median(depth_gt) / np.median(depth_pred)
        depth_pred[depth_pred > 70] = 70
        errors.append(compute_errors(depth_gt, depth_pred))
    mean_errors = np.mean(errors, 0)

print(("{:>8} | " * 4).format("abs_rel", "sq_rel", "rmse", "rmse_log"))
print(("{: 8.3f} , " * 4).format(*mean_errors.tolist()))`
Thank you very much for your patient answer, I got similar results with this code.
![1111111](https://github.com/user-attachments/assets/86f7bd6d-b0f2-4699-be05-2f1bbdc3de2e)

**Comment by GuanghuiWei** [comment]
`
from layers import disp_to_depth
import networks
import cv2
import os
import torch
from scipy import io
import numpy as np

load_weights_folder = '/path/to/weight/'
main_path = 'path/to/mask_3d/'
encoder_path = os.path.join(load_weights_folder, "encoder.pth")
decoder_path = os.path.join(load_weights_folder, "depth.pth")
encoder_dict = torch.load(encoder_path)
encoder = networks.LiteMono(model='lite-mono',
                            height=encoder_dict['height'],
                            width=encoder_dict['width'])
depth_decoder = networks.DepthDecoder(encoder.num_ch_enc, scales=range(3))

model_dict = encoder.state_dict()
encoder.load_state_dict({k: v for k, v in encoder_dict.items() if k in model_dict})
depth_decoder.load_state_dict(torch.load(decoder_path), strict=False)

encoder.cuda()
encoder.eval()
depth_decoder.cuda()
depth_decoder.eval()


def compute_errors(gt, pred):
    rmse = (gt - pred) ** 2
    rmse = np.sqrt(rmse.mean())

    rmse_log = (np.log10(gt) - np.log10(pred)) ** 2
    rmse_log = np.sqrt(rmse_log.mean())

    abs_rel = np.mean(np.abs(gt - pred) / gt)

    sq_rel = np.mean(((gt - pred) ** 2) / gt)

    return abs_rel, sq_rel, rmse, rmse_log


with open(os.path.join('/path/to/make3d_test/', "test_files.txt")) as f:
    test_filenames = f.read().splitlines()


depths_gt = []
images = []
ratio = 2
h_ratio = 1 / (1.33333 * ratio)
color_new_height = int(1704 / 2)
depth_new_height = 21
for filename in test_filenames:
    mat = io.loadmat(os.path.join(main_path, "Gridlaserdata", "depth_sph_corr-{}.mat".format(filename)))
    depths_gt.append(mat["Position3DGrid"][:, :, 3])

    image = cv2.imread(os.path.join(main_path, "Test134", "img-{}.jpg".format(filename)))
    image = image[int((2272 - color_new_height) / 2):int((2272 + color_new_height) / 2), :, :]
    images.append(image[:, :, ::-1])
depths_gt_resized = map(lambda x: cv2.resize(x, (305, 407), interpolation=cv2.INTER_NEAREST), depths_gt)
depths_gt_cropped = map(lambda x: x[int((55 - 21) / 2):int((55 + 21) / 2), :], depths_gt)

depths_gt_cropped = list(depths_gt_cropped)
errors = []
with torch.no_grad():
    for i in range(len(images)):
        input_color = images[i]
        input_color = cv2.resize(input_color / 255.0, (640, 192), interpolation=cv2.INTER_NEAREST)

        input_color = torch.tensor(input_color, dtype=torch.float).cuda().permute(2, 0, 1)[None, :, :, :]

        output = depth_decoder(encoder(input_color))
        pred_disp, _ = disp_to_depth(output[("disp", 0)], 1e-3, 70)
        pred_disp = pred_disp.squeeze().cpu().numpy()
        depth_gt = depths_gt_cropped[i]
        depth_pred = 1 / pred_disp
        depth_pred = cv2.resize(depth_pred, depth_gt.shape[::-1], interpolation=cv2.INTER_NEAREST)
        mask = np.logical_and(depth_gt > 0, depth_gt < 70)
        depth_gt = depth_gt[mask]
        depth_pred = depth_pred[mask]
        depth_pred *= np.median(depth_gt) / np.median(depth_pred)
        depth_pred[depth_pred > 70] = 70
        errors.append(compute_errors(depth_gt, depth_pred))
    mean_errors = np.mean(errors, 0)

print(("{:>8} | " * 4).format("abs_rel", "sq_rel", "rmse", "rmse_log"))
print(("{: 8.3f} , " * 4).format(*mean_errors.tolist()))
`

Thank you very much for your patient answer, I got similar results with this code.
![1111111](https://github.com/user-attachments/assets/c8aead88-4a9e-42ff-a681-475cabfadf19)

**Comment by noahzn** [comment]
@Wshenv Good to know!!
